CMakeLists.txt

```diff
@@ -533,28 +515,28 @@ foreach(header ${HEADERS_TO_CHECK})
     string(TOUPPER ${header} tmp)
 
     string(REGEX REPLACE [./] "_" upper_header ${tmp})
-    check_include_file(${header} HAVE_${upper_header})
+    set(header_flag "HAVE_${upper_header}")
+    check_include_file(${header} ${header_flag})
+    if (header_flag)
+        add_compile_definitions(${header_flag})
+    endif()
```

Comment:
We can add the compile definitions here instead of doing it individually (so we don't need to set e.g. `HAVE_DIRENT_H`)